b2sdk/sync/policy.py

```diff
@@ -88,17 +89,20 @@ def _should_transfer(self):
         """
         Decide whether to transfer the file from the source to the destination.
         """
-        if self._source_file is None or self._source_file.latest_version().action == 'hide':
+        if self._source_path is None or (
+            self._source_path.type_() == PathType.B2 and
```

Comment:
we did not check for type here, this is a hidden `isinstance()` and I don't think we really need it, do we?

I think you check it to make sure we won't get an exception for just touching `.action`, while what should happen here is that `_source_path.is_visible()`, where single-version local path will just return True and remote will check `.latest_version().action == 'hide'`